dask_planner/src/sql/logical/join.rs

```diff
@@ -20,36 +20,29 @@ pub struct PyJoin {
 impl PyJoin {
     #[pyo3(name = "getCondition")]
     pub fn join_condition(&self) -> PyExpr {
-        // TODO: This logic should be altered once https://github.com/apache/arrow-datafusion/issues/2496 is complete
-        if self.join.on.len() >= 1 {
-            let (left_col, right_col) = &self.join.on[0];
-            let mut root_expr: Expr = Expr::BinaryExpr {
-                left: Box::new(Expr::Column(left_col.clone())),
-                op: Operator::Eq,
-                right: Box::new(Expr::Column(right_col.clone())),
-            };
-            for idx in 1..self.join.on.len() {
-                let (left_col, right_col) = &self.join.on[idx];
-                let ex: Expr = Expr::BinaryExpr {
-                    left: Box::new(Expr::Column(left_col.clone())),
-                    op: Operator::Eq,
-                    right: Box::new(Expr::Column(right_col.clone())),
-                };
+        // equi-join filters
+        let mut filters: Vec<Expr> = self
+            .join
+            .on
+            .iter()
+            .map(|(l, r)| Expr::Column(l.clone()).eq(Expr::Column(r.clone())))
+            .collect();
 
-                root_expr = Expr::BinaryExpr {
-                    left: Box::new(root_expr),
-                    op: Operator::Eq,
-                    right: Box::new(ex),
-                }
-            }
-            PyExpr::from(
-                root_expr,
-                Some(vec![self.join.left.clone(), self.join.right.clone()]),
-            )
-        } else {
-            panic!("Join Length: {}, Encountered a Join with more than a single column for the join condition. This is not currently supported
-            until DataFusion makes some changes to allow for Joining logic other than just Equijoin.", self.join.on.len())
+        // other filter conditions
+        if let Some(filter) = &self.join.filter {
+            filters.push(filter.clone());
         }
+
+        assert!(filters.len() > 0);
```

Comment:
Is this a safe assumption? Does this method get called if there is a join with no join condition? If so, should it return `Option<PyExpr>`?